tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -170,6 +171,20 @@ public void Inject<TCarrier>(SpanContext context, TCarrier carrier, Action<TCarr
                         if (localSpanContext.RawTraceId == spanContext.RawTraceId)
```

Comment:
This if statement is the source of my suggestions to add more testing for other propagators. Say for example that `localSpanContext` was initialized due to reading Datadog headers. We only want to run the if block when we are currently running the `tracecontext` propagator (e.g. configuration is `datadog,tracecontext`) but not when we have are running the `b3multi` propagator (e.g. configuration is `datadog,b3multi`). I think we'll need to add another clause to the if statement to make sure we're running the `tracecontext` propagator

Reply:
Did some changes based on the configuration below: https://github.com/DataDog/dd-trace-dotnet/pull/5721/commits/aaac0d0d30130b6723f8528bf209055b09569792 but then noticed the RFC only mentions from Datadog headers specifically so may need to change it to match that, WDYT?
https://docs.google.com/document/d/1Zc7uAQTJ2G5vHEbmx8k4vMnYdQ-BnqZxewXk0ohgd3g/edit#heading=h.hin7r1gouzj

>Fallback 1: _dd.parent_id SHOULD be set using the parent_id extracted from datadog (x-datadog-parent-id).